### DISCUSSION TITLE
Difference in __mocks__ automocks with Jest

### DISCUSSION THREAD

**Discussion by EvHaus** [vitest-dev/vitest#466] [Q&A] [answered]
In Jest, when you have automocks like this:

- `__mocks__`
    - `ora`
        - `index.js`

This mock gets auto loaded for all tests. However, in Vitest it only gets automocked if you also add `vi.mock('ora')` in your tests.

Curious, is this by design? In some ways I like it because it's more explicit, but it does make migrating from Jest to Vitest more difficult.

**Comment by sheremet-va** [comment, marked as answer]
This was done by design. We don't support automockig at this stage of development. Two main reasons are: it's more explicit and it gives better performance, since we don't need to resolve every request and try to find a mock.

I personally don't think we should support automocking. I asked people in Discord, when implemented `vi.mock`, and the people voted to not includid right now. If you think it's needed, you can start a discussion in Discord 👍🏻

**Reply by EvHaus** [reply]
I'm happy with this design. I would recommend updating the docs though. Perhaps add:

> Unlike Jest, files in `__mocks__` are not loaded unless `vi.mock()` is called.

in the ["Automocking algorithm" section?](https://vitest.dev/guide/mocking-modules.html#automocking-algorithm)

**Reply by sheremet-va** [reply]
PR is welcomed :)
